### PR TITLE
Nfdiv 991 and nfdiv 992 application issued notifications

### PR DESCRIPTION
**Pull request by lara-mpt**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/NFDIV-991
https://tools.hmcts.net/jira/browse/NFDIV-992

Change description
Citizen email notifications and state change for the caseworker issue application events

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No

old pr - https://github.com/hmcts/nfdiv-case-api/pull/648 
(had to close and reopen pr as build was broken)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendAosPack.java [hunk 1]
```diff
@@ -21,24 +21,27 @@
     @Override
     public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
 
-        final Long caseId = caseDetails.getId();
-        final CaseData caseData = caseDetails.getData();
+        if (caseDetails.getData().getApplication().isSolicitorApplication()) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendAosPack.java:24]: may be we can combine the two ifs ? one on line 24 and 29

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java [hunk 2]
```diff
@@ -33,27 +33,30 @@
...
-            caseDataDocumentService.renderDocumentAndUpdateCaseData(
-                caseData,
-                DOCUMENT_TYPE_RESPONDENT_INVITATION,
-                templateContentSupplier,
-                caseId,
-                CITIZEN_RESP_AOS_INVITATION,
-                caseData.getApplicant1().getLanguagePreference(),
-                RESP_AOS_INVITATION_DOCUMENT_NAME + caseId
-            );
+        if (caseDetails.getData().getApplication().isSolicitorApplication()) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java:36]: this if can combined with one on line 42 ?

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateMiniApplication.java [hunk 3]
```diff
@@ -31,23 +31,26 @@
...
-            caseData,
-            DIVORCE_APPLICATION,
-            templateContentSupplier,
-            caseId,
-            DIVORCE_MINI_APPLICATION,
-            caseData.getApplicant1().getLanguagePreference(),
-            DIVORCE_MINI_APPLICATION_DOCUMENT_NAME + caseId
-        );
+        if (caseDetails.getData().getApplication().isSolicitorApplication()) {
+
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateMiniApplication.java:35]: would be nice to add some logs here

Comment (lara-mpt) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateMiniApplication.java:35]: @nitinprabhu I'm not sure what else to add other than line 40?

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateMiniApplication.java:35]: may be that's fine I was thinking something like log.info("Application with case id {} is solicitor application, caseId);

File: src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerIssueApplicationFT.java [hunk 4]
```diff
@@ -34,7 +42,24 @@ public void shouldUpdateCaseDataWhenAboutToSubmitCallbackIsSuccessful() throws E
...
+
+    @Test
+    public void shouldUpdateCaseDataAndSendEmailsWhenAboutToSubmitCallbackIsSuccessfulForCitizenApplication() throws Exception {
+        final Map<String, Object> caseData = caseData(CITIZEN_REQUEST);
+        final Response response = triggerCallback(caseData, CASEWORKER_ISSUE_APPLICATION, ABOUT_TO_SUBMIT_URL);
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        // document_url and document_binary_url are ignored using ${json-unit.ignore}
+        // assertion will fail if the above elements are missing actual value
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerIssueApplicationFT.java:57]: does this need to be removed ?

**Review by nitinprabhu — APPROVED** [review]

**Pull request closed without merging**